low-level/sensors/impl/generic/node_hw.py

```diff
@@ -480,6 +481,54 @@ def run(self):
             logger.warning(f"{self.SENSOR_NAME} Node hw monitoring disabled")
             self.shutdown()
 
+    def _check_missmatch_resource_status(self):
+        for sensor_id in self.faulty_resources:
+            if self.faulty_resources[sensor_id]['resource'] in ['fan', 'disk', 'temperature', 'voltage']:
+                _, sensor_get_data, _ = self._get_sensor_props(sensor_id)
+                if sensor_get_data and 'Assertion Events' in sensor_get_data \
+                        and 'Status' in sensor_get_data:
+                    # 'Status': 'ok'
+                    # 'Assertion Events': ''
+                    if sensor_get_data['Assertion Events'] in [""] \
+                        and sensor_get_data['Status'] == 'ok':
+                        self._create_sel_event_entry(sensor_id)
+
+            else:
+                _, sdr_get_data = self._get_sensor_sdr_props(sensor_id)
+                if sdr_get_data and 'States Asserted' in sdr_get_data:
+                    #  'States Asserted': 'Power Supply, Presence detected'
+                    resource_state = re.sub(',  +', ', ', re.sub('[\[\]]','',
+                        sdr_get_data['States Asserted']).replace('\n',','))
+                    sensor_status = resource_state.split(',')
+                    sensor_status = sensor_status[0] if len(sensor_status) == 1 else sensor_status[1]
+                    sensor_status = sensor_status.strip()
+                    if sensor_status in ["", "Presence detected"]:
+                        self._create_sel_event_entry(sensor_id)
+
+    def _create_sel_event_entry(self, sensor_id):
+        event_map_dict = {
+            'Lower Non-Recoverable': 'lnr',
+            'Lower Critical': 'lcr',
+            'Lower Non-Critical': 'lnc',
+            'Upper Non-Critical': 'unc',
+            'Upper Critical': 'ucr',
+            'Upper Non-Recoverable': 'unr'
+        }
```

Comment:
These events are specific to 'Fan' only.
How other fru events are handled?

Reply:
We have possible mapping events for FAN, Temperature, voltage.
[root@sm17-r18 satish]# ipmitool event FAN1
Finding sensor FAN1... ok
Sensor States:
  lnr : Lower Non-Recoverable
  lcr : Lower Critical
  lnc : Lower Non-Critical
  unc : Upper Non-Critical
  ucr : Upper Critical
  unr : Upper Non-Recoverable
  
and for resources no mapping needed.
[root@sm17-r18 satish]# ipmitool event 'PS1 Status'
Finding sensor PS1 Status... ok
Sensor States:
  Presence detected
  Failure detected
  Predictive failure
  Power Supply AC lost
  AC lost or out-of-range
  AC out-of-range, but present
  Config Error
  Power Supply Inactive